src/hyperframe/frame.py

```diff
@@ -68,25 +68,52 @@ def __init__(self, stream_id, flags=()):
 
         if (not self.stream_id and
            self.stream_association == _STREAM_ASSOC_HAS_STREAM):
-            raise InvalidDataError('Stream ID must be non-zero')
+            raise InvalidDataError(
+                'Stream ID must be non-zero for {}'.format(
+                    type(self).__name__,
+                )
+            )
         if (self.stream_id and
            self.stream_association == _STREAM_ASSOC_NO_STREAM):
-            raise InvalidDataError('Stream ID must be zero')
+            raise InvalidDataError(
+                'Stream ID must be zero for {} with stream_id: {}'.format(
+                    type(self).__name__,
+                    self.stream_id,
+                )
+            )
 
     def __repr__(self):
-        flags = ", ".join(self.flags) or "None"
-        body = binascii.hexlify(self.serialize_body()).decode('ascii')
-        if len(body) > 20:
-            body = body[:20] + "..."
         return (
-            "{type}(Stream: {stream}; Flags: {flags}): {body}"
+            "{}(stream_id: {}; flags: {}): {}"
         ).format(
-            type=type(self).__name__,
-            stream=self.stream_id,
-            flags=flags,
-            body=body,
+            type(self).__name__,
+            self.stream_id,
+            ", ".join(sorted(self.flags)) or "None",
```

Comment:
And then just this here:
```suggestion
            repr(self.flags),
```

Reply:
This was actually still missing - so I added it!